frontend_tests/zjsunit/parallel_runner.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const child_process = require("child_process");
+const path = require("path");
+
+const parallel = process.argv[2];
+
+// Find the files we need to run.
+const files = process.argv.slice(3);
+if (files.length === 0) {
+    throw new Error("No tests found");
+}
+
+const index = path.join(__dirname, "index.js");
+
+function create_file_groups(files, parallel) {
+    const group_size = Math.ceil(files.length / parallel);
+    const file_groups = [];
+    for (let i = 0; i < parallel; i += 1) {
+        const file_group = files.slice(i * group_size, (i + 1) * group_size);
+        file_groups.push(file_group);
+    }
+    return file_groups;
+}
+
+const file_groups = create_file_groups(files, parallel);
+const worker_processes = [];
+for (let i = 0; i < parallel; i += 1) {
+    const worker_process = child_process.spawn("node", [index, ...file_groups[i]]);
```

Comment:
[`child_process.fork`](https://nodejs.org/api/child_process.html#child_processforkmodulepath-args-options) is cleaner for this case.